File: src/Entity/Historique.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;


#[ORM\Entity]
class Historique
{

    #[ORM\Id]
    #[ORM\Column(type: "integer")]
    private int $id;

    #[ORM\Column(type: "string", length: 50)]
    private string $action;

    #[ORM\Column(type: "string", length: 50)]
    private string $entite;

    #[ORM\Column(type: "datetime")]
    private \DateTimeInterface $date;

    #[ORM\Column(type: "text")]
    private string $details;

    public function getId()
    {
        return $this->id;
    }

    public function setId($value)
    {
        $this->id = $value;
    }

    public function getAction()
    {
        return $this->action;
    }

    public function setAction($value)
    {
        $this->action = $value;
    }

    public function getEntite()
    {
        return $this->entite;
    }

    public function setEntite($value)
    {
        $this->entite = $value;
    }

    public function getDate()
    {
        return $this->date;
    }

    public function setDate($value)
    {
        $this->date = $value;
    }

    public function getDetails()
    {
        return $this->details;
    }

    public function setDetails($value)
    {
        $this->details = $value;
    }
}

File: src/main/java/com/esprit/models/Historique.java
package com.esprit.models;

import java.sql.Timestamp;

public class Historique {
    private int id;
    private String action;
    private String entite;
    private Timestamp date;
    private String details;

    // Constructeurs
    public Historique() {}

    public Historique(int id, String action, String entite, Timestamp date, String details) {
        this.id = id;
        this.action = action;
        this.entite = entite;
        this.date = date;
        this.details = details;
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getAction() {
        return action;
    }

    public void setAction(String action) {
        this.action = action;
    }

    public String getEntite() {
        return entite;
    }

    public void setEntite(String entite) {
        this.entite = entite;
    }

    public Timestamp getDate() {
        return date;
    }

    public void setDate(Timestamp date) {
        this.date = date;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }

    @Override
    public String toString() {
        return "Historique{" +
                "id=" + id +
                ", action='" + action + '\'' +
                ", entite='" + entite + '\'' +
                ", date=" + date +
                ", details='" + details + '\'' +
                '}';
    }
}
